tests/common/plugins/pdu_controller/__init__.py

```diff
@@ -30,15 +30,28 @@ def get_pdu_visible_vars(inventories, pdu_hostnames):
     return pdu_hosts_vars
 
 
+def get_sup_node_or_first_node(duthosts):
+    # accomodate for T2 chassis, which only SUP has pdu info
+    # single-dut get itself
+    if len(duthosts) == 1:
+        return duthosts[0]
+    # try to find sup node in multi-dut
+    for dut in duthosts:
+        if dut.is_supervisor_node():
+            return dut
+    # if not chassis, it's dualtor, return first node
+    return duthosts[0]
```

Comment:
should be return random host here instead of always returning first dut?

Reply:
this is how enum_supervisor_duthost is implemented